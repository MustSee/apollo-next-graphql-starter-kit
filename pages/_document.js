import Document, { Head, Main, NextScript } from 'next/document';
import React from 'react';

class MyDocument extends Document {

  render() {
    return (
      <html lang={'en'}>
        <Head>
          <title>High performance web frameswork</title>
          <meta charSet="utf-8" />
          <meta name="viewport" content="initial-scale=1.0, width=device-width" />
          <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/antd/3.0.1/antd.min.css" />
          <link rel="icon" href="favicon.ico" />
        </Head>
        <body>
          <Main />
          <NextScript />
        </body>
      </html>
    );
  }
}

export default MyDocument;
