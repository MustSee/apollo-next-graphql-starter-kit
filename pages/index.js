import React from 'react';
import withApp from '../lib/withApp';
import UserTable from '../components/UserTable/UserTable';
import { Layout, Title, Row } from '../style/index';

export const HomePage = () => (<Layout>
  <Row>
    <Title {...{ type: 'h1' }}>High performance web starter-kit</Title>
  </Row>
  <Row>
    <UserTable />
  </Row>
</Layout>);

export default withApp(HomePage);
