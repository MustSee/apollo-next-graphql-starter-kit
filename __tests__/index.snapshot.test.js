import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { HomePage } from '../pages/index';

const { test, expect } = global;

test('Homepage component shallow renderes its content correctly', () => {
  const snapshotComponent = shallow(
    <HomePage />,
    );
  expect(toJson(snapshotComponent)).toMatchSnapshot();
});
