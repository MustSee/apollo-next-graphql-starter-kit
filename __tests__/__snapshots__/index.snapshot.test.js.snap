// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage component shallow renderes its content correctly 1`] = `
<LayoutProvider>
  <p>
    High performance web frameswork
  </p>
  <Apollo(ApolloComponent) />
</LayoutProvider>
`;
